sdks/python/apache_beam/io/hadoopfilesystem.py

```diff
@@ -163,19 +181,25 @@ def join(self, base_url, *paths):
     Returns:
       Full url after combining all the passed components.
     """
-    basepath = self._parse_url(base_url)
-    return _HDFS_PREFIX + self._join(basepath, *paths)
+    server, basepath = self._parse_url(base_url)
+    # TODO full_urls check and test
+    return _HDFS_PREFIX + self._join(server, basepath, *paths)
 
   def _join(self, basepath, *paths):
     return posixpath.join(basepath, *paths)
 
   def split(self, url):
-    rel_path = self._parse_url(url)
+    server, rel_path = self._parse_url(url)
+    if server is None:
+      server = ''
+    else:
+      server = '/' + server
```

Comment:
Is this only expected to ever work in posix? If not, should we use `os.path.join`?

Reply:
Nevermind. I think if we consider hfds as a constraint this is fine.